### PR TITLE
Euclidean recursive method + doctests + type hints

### PR DESCRIPTION
**Pull request by mateuszz0000**
### **Describe your change:**
Added recursive method for Euclidean algorithm.
Added doctests and type hints for existing implementation.

* [x] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: other/euclidean_gcd.py [hunk 1]
```diff
@@ -1,14 +1,39 @@
...
+    >>> euclidean_gcd(3, 5)
+    1
+
+    >>> euclidean_gcd(6, 3)
+    3
+    """
     while b:
         t = b
         b = a % b
         a = t
```
Comment (cclauss) [inline other/euclidean_gcd.py:16]:
```suggestion
        a, b = b, a % b
```
Multiple assignment in Python eliminates the need for these temporary variables.

File: other/euclidean_gcd.py [hunk 2]
```diff
@@ -1,14 +1,39 @@
...
+    >>> euclidean_gcd_recursive(3, 5)
+    1
+
+    >>> euclidean_gcd_recursive(6, 3)
+    3
+    """
+    if b == 0:
+        return a
+    else:
+        return euclidean_gcd_recursive(b, a % b)
```
Comment (cclauss) [inline other/euclidean_gcd.py:34]:
```suggestion
    return a if b == 0 else euclidean_gcd_recursive(b, a % b)
```

File: other/euclidean_gcd.py [hunk 3]
```diff
@@ -1,14 +1,39 @@
...
+
+
 def main():
     print("GCD(3, 5) = " + str(euclidean_gcd(3, 5)))
     print("GCD(5, 3) = " + str(euclidean_gcd(5, 3)))
```
Comment (cclauss) [inline other/euclidean_gcd.py:39]:
```suggestion
    print(f"euclidean_gcd(3, 5) = {euclidean_gcd(3, 5)}")
    print(f"euclidean_gcd(5, 3) = {euclidean_gcd(5, 3)}")
```

Comment (TheSuperNoob) [reply other/euclidean_gcd.py:39]: Might as well include the remaining 3 print statements below this as well

File: other/euclidean_gcd.py (hunk 1 above)
Comment (mateuszz0000) [reply other/euclidean_gcd.py]: Fixed

File: other/euclidean_gcd.py (hunk 2 above)
Comment (mateuszz0000) [reply other/euclidean_gcd.py]: Fixed as well :)

File: other/euclidean_gcd.py (hunk 3 above)
Comment (mateuszz0000) [reply other/euclidean_gcd.py]: Thanks!

**Comment by mateuszz0000** [comment]
Is something to do more here?

**Pull request merged**